aws-ssm-maintenancewindow/src/main/java/software/amazon/ssm/maintenancewindow/DeleteHandler.java

```diff
@@ -1,27 +1,96 @@
 package software.amazon.ssm.maintenancewindow;
 
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.awssdk.services.ssm.model.DeleteMaintenanceWindowRequest;
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
 import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
 import software.amazon.cloudformation.proxy.Logger;
-import software.amazon.cloudformation.proxy.ProgressEvent;
 import software.amazon.cloudformation.proxy.OperationStatus;
-import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.ssm.maintenancewindow.translator.ExceptionTranslator;
+import software.amazon.ssm.maintenancewindow.util.ClientBuilder;
+
+import java.util.Optional;
 
 public class DeleteHandler extends BaseHandler<CallbackContext> {
 
+    private static final SsmClient SSM_CLIENT = ClientBuilder.getClient();
+    private final ExceptionTranslator exceptionTranslator;
+
+    DeleteHandler() {
+        this.exceptionTranslator = new ExceptionTranslator();
+    }
+
+    /**
+     * Used for unit tests.
+     *
+     * @param exceptionTranslator Used for translating service model exceptions.
+     */
+    DeleteHandler(final ExceptionTranslator exceptionTranslator) {
+        this.exceptionTranslator = exceptionTranslator;
+    }
+
     @Override
     public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final Logger logger) {
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+
+        logger.log(String.format("Processing DeleteHandler request %s", request));
 
         final ResourceModel model = request.getDesiredResourceState();
 
-        // TODO : put your code here
+        final ProgressEvent<ResourceModel, CallbackContext> progressEvent =
+                ProgressEvent.<ResourceModel, CallbackContext>builder()
+                        .resourceModel(model)
+                        .build();
+
+        progressEvent.setStatus(OperationStatus.FAILED);
+
+        final Optional<DeleteMaintenanceWindowRequest.Builder> optionalRequestBuilder = initializeRequestBuilder(model);
```

Comment:
Maybe we can do sth similar as UpdateHandler here:
if (StringUtils.isNullOrEmpty(windowId)) {
             progressEvent.setErrorCode(HandlerErrorCode.InvalidRequest);
             progressEvent.setMessage("WindowId must be present to read the existing association.");
             return progressEvent;
}

Reply:
Addressed